dokka-runners/runner-gradle-plugin-classic/build.gradle.kts

```diff
@@ -7,6 +7,7 @@ import dokkabuild.overridePublicationArtifactId
 
 plugins {
     id("dokkabuild.gradle-plugin")
+    id("dokkabuild.dev-maven-publish")
```

Comment:
The test tasks in the DGP subproject aren't integration tests, and don't need the dev-maven repo. `dokkabuild.dev-maven-publish` is applied in this subproject so that the integration test project can fetch the `dev-maven` repository artifacts.

Does my explanation make sense? :)

Reply:
Honestly, it is still not clear for me. Do we add `dokkabuild.dev-maven-publish` to fill [devMavenRepositories](https://github.com/Kotlin/dokka/blob/53a170418be255131981888a250244f626b044ac/build-logic/src/main/kotlin/dokkabuild/DevMavenPublishExtension.kt#L18) with local repository of `runner-gradle-plugin-classic` ?

Reply:
Yes, that's right. 

It might make it clearer to think about `dokkabuild.dev-maven-publish` as having two usages:

1. The first usage is to share a project-local Maven repo.

   It does this by adding PublishToMavenLocal tasks, and shares the output of these tasks with a Configuration (named `devPublicationConsumable`).

2. The second usage is to consume project-local Maven repos from dependencies added to the `devPublication` Configuration (and then `devMavenPublish.configureTask(this)` adds those repos as a task-input).


So we add `dokkabuild.dev-maven-publish` to `runner-gradle-plugin-classic` to add a PublishToMavenLocal task, and also add `dokkabuild.dev-maven-publish` to the integration tests to resolve the result.

Does that help? :)